CodeSnippetsReflection.OpenAPI/LanguageGenerators/TypeScriptGenerator.cs

```diff
@@ -20,40 +17,183 @@ public class TypeScriptGenerator : ILanguageGenerator<SnippetModel, OpenApiUrlTr
         /// <param name="snippetModel">Model of the Snippets info <see cref="SnippetModel"/></param>
         /// <returns>String of the snippet in TypeScript code</returns>
         ///
-
         private const string ClientVarName = "graphServiceClient";
-        private const string ClientVarType = "GraphServiceClient";
+
         private const string RequestHeadersVarName = "headers";
         private const string RequestOptionsVarName = "options";
         private const string RequestConfigurationVarName = "configuration";
         private const string RequestParametersVarName = "queryParameters";
         private const string RequestBodyVarName = "requestBody";
 
+        private static readonly Regex FunctionRegex =
+            new Regex(@"(\w+)\(([^)]*)\)", RegexOptions.Compiled, TimeSpan.FromMilliseconds(200));
+
+        private static readonly Regex ParamRegex =
+            new Regex(@"(\w+)\s*=\s*'[^']*'", RegexOptions.Compiled, TimeSpan.FromMilliseconds(200));
+
         public string GenerateCodeSnippet(SnippetModel snippetModel)
         {
             if (snippetModel == null) throw new ArgumentNullException("Argument snippetModel cannot be null");
 
             var codeGraph = new SnippetCodeGraph(snippetModel);
             var snippetBuilder = new StringBuilder(
-                                    "//THIS SNIPPET IS A PREVIEW FOR THE KIOTA BASED SDK. NON-PRODUCTION USE ONLY" + Environment.NewLine +
-                                    $"const {ClientVarName} = {ClientVarType}.init({{authProvider}});{Environment.NewLine}{Environment.NewLine}");
+                "//THIS SNIPPET IS A PREVIEW FOR THE KIOTA BASED SDK. NON-PRODUCTION USE ONLY" + Environment.NewLine);
 
+            writeImportStatements(codeGraph, snippetBuilder);
             writeSnippet(codeGraph, snippetBuilder);
 
             return snippetBuilder.ToString();
         }
 
+        private static void writeImportStatements(SnippetCodeGraph codeGraph, StringBuilder builder)
```

Comment:
Conventionally, this should start with an uppercase (PascalCase). https://learn.microsoft.com/en-us/dotnet/csharp/fundamentals/coding-style/identifier-names#pascal-case
```suggestion
        private static void WriteImportStatements(SnippetCodeGraph codeGraph, StringBuilder builder)
```